src/core/src/op/space_to_batch.cpp

```diff
@@ -64,84 +62,67 @@ void op::v1::SpaceToBatch::validate_and_infer_types() {
+bool SpaceToBatch::visit_attributes(AttributeVisitor& visitor) {
     OV_OP_SCOPE(v1_SpaceToBatch_visit_attributes);
     return true;
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-bool ngraph::op::v1::SpaceToBatch::evaluate_space_to_batch(const HostTensorVector& outputs,
-                                                           const HostTensorVector& inputs) const {
-    if (outputs[0]->get_partial_shape().is_dynamic()) {
-        std::vector<ov::PartialShape> input_shapes;
-        input_shapes.reserve(inputs.size());
-
-        for (size_t i = 0; i < inputs.size(); ++i) {
-            input_shapes.push_back(inputs[i]->get_partial_shape());
-            if (input_shapes.back().is_dynamic()) {
-                return false;
-            }
-        }
-
-        const auto output_shape = shape_infer(this, input_shapes, make_tensor_accessor(inputs)).front().to_shape();
-
-        outputs[0]->set_element_type(inputs[0]->get_element_type());
-        outputs[0]->set_shape(output_shape);
-    }
-
+namespace space_to_batch {
+namespace {
+bool evaluate(TensorVector& outputs, const TensorVector& inputs) {
     const auto& data = inputs[0];
     const auto& out = outputs[0];
-    size_t elem_size = data->get_element_type().size();
+    const auto elem_size = data.get_element_type().size();
 
-    auto data_shape = data->get_shape();
+    auto data_shape = data.get_shape();
 
-    if (!(data->get_shape().size() == 3 || data->get_shape().size() == 4 || data->get_shape().size() == 5)) {
+    if (!(data.get_shape().size() == 3 || data.get_shape().size() == 4 || data.get_shape().size() == 5)) {
         return false;
     }
 
-    size_t block_values_size = shape_size(inputs[1]->get_shape());
-    const auto* block_values = inputs[1]->get_data_ptr<int64_t>();
-    const auto* pads_begin = inputs[2]->get_data_ptr<int64_t>();
-    const auto* pads_end = inputs[3]->get_data_ptr<int64_t>();
+    const auto block_values_size = shape_size(inputs[1].get_shape());
+    const auto block_values = static_cast<const int64_t*>(inputs[1].data());
+    const auto pads_begin = static_cast<const int64_t*>(inputs[2].data());
+    const auto pads_end = static_cast<const int64_t*>(inputs[3].data());
 
     const char* pad_value = nullptr;
     const std::vector<char> pad_zero_value(elem_size, 0);
     if (inputs.size() == 4) {
-        pad_value = inputs[3]->get_data_ptr<char>();
+        pad_value = static_cast<const char*>(inputs[3].data());
     } else {
         pad_value = pad_zero_value.data();
     }
-    CoordinateDiff pads_begin_vec(shape_size(inputs[2]->get_shape()));
-    pads_begin_vec.assign(pads_begin, pads_begin + shape_size(inputs[2]->get_shape()));
-    CoordinateDiff pads_end_vec(shape_size(inputs[2]->get_shape()));
-    pads_end_vec.assign(pads_end, pads_end + shape_size(inputs[2]->get_shape()));
+    CoordinateDiff pads_begin_vec(shape_size(inputs[2].get_shape()));
+    pads_begin_vec.assign(pads_begin, pads_begin + shape_size(inputs[2].get_shape()));
```

Comment:
Is out of scope regardging migration but maybe worth check.

```suggestion
    CoordinateDiff pads_begin_vec(pads_begin, pads_begin + shape_size(inputs[2].get_shape()));
```
Will it provide same result?

Calculate `shape_size(inputs[2].get_shape())` before pads creation and use twice instead calculate twice?

Reply:
Changed as suggested.